helm-core.el

```diff
@@ -5121,16 +5112,17 @@ without recomputing them, it should be a list of lists."
              (member (assoc-default 'name source) helm-source-filter))
          (>= len
              (helm-aif (assq 'requires-pattern source) (or (cdr it) 1) 0))
-         ;; Entering repeatedly these strings (*, ?) takes 100% CPU
-         ;; and hang emacs on MacOs preventing deleting backward those
-         ;; characters (Bug#1802). Update: it seems it is no more true,
-         ;; thus this affect bug#2423, so let's remove this for now.
-         ;; (not (string-match-p "\\`[*]+\\'" helm-pattern))
-         ;; These incomplete regexps hang helm forever
-         ;; so defer update. Maybe replace spaces quoted when using
-         ;; multi-match.
-         (not (member (replace-regexp-in-string "\\s\\ " " " helm-pattern)
-                      helm-update-blacklist-regexps)))))
+         ;; bugs were mentioned here before: Bug#1802, bug#2423
+         ;; must be fixed already by 'helm-re-search-forward'
+         ;;
+         ;; short patterns that are alone not expected to return any results,
+         ;; ignore in order to not erase Helm buffer on such patterns.
+         (not
+          (let ((split-string-default-separators "[ \f\t\n\r\v]+")
+                (subpatterns (string-split helm-pattern)))
```

Comment:
Your split-string is not affected by split-string-default-separators because of let and not let*.
BTW why do you want such a separator? isn't splitting with helm-mm-split-pattern enough?

Reply:
> Your split-string is not affected by split-string-default-separators because of let and not let*.

Yeah, added it and then got distracted by some other aspect. That binding probably not needed at all. Chances that something will change separator definitions are almost nonexistent.